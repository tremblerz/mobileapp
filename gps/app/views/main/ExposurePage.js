import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Dimensions, ImageBackground, StatusBar, View } from 'react-native';
import { SvgXml } from 'react-native-svg';

import backgroundImage from '../../../../shared/assets/images/backgroundAtRisk.png';
import StateAtRisk from '../../../../shared/assets/svgs/stateAtRisk';
import { Button } from '../../components/Button';
import { Typography } from '../../components/Typography';
import { Theme } from '../../constants/themes';
import { MayoButton } from './MayoButton';
import { styles } from './style';

export const ExposurePage = () => {
  const { t } = useTranslation();
  const navigation = useNavigation();
  const buttonLabel = t('label.see_exposure_history');
  const size = Dimensions.get('window').height;

  return (
    <Theme use='charcoal'>
      <ImageBackground source={backgroundImage} style={styles.backgroundImage}>
        <StatusBar
          barStyle='light-content'
          backgroundColor='transparent'
          translucent
        />
        <View style={styles.pulseContainer}>
          <SvgXml
            xml={StateAtRisk}
            width={size ? size : 80}
            height={size ? size : 80}
          />
        </View>

        <View style={styles.mainContainer}>
          <View style={styles.contentAbovePulse}>
            <Typography style={styles.mainTextAbove}>
              {t('label.home_at_risk_header')}
            </Typography>
            <Typography style={styles.subsubheaderText}>
              {t('label.home_at_risk_subsubtext')}
            </Typography>
          </View>
          <View style={styles.contentBelowPulse}>
            <Typography style={styles.subheaderText}>
              {t('label.home_at_risk_subtext')}
            </Typography>
            <View style={styles.buttonContainer}>
              <Button
                label={buttonLabel}
                onPress={() => navigation.navigate('ExposureHistoryScreen')}
                style={styles.buttonContainer}
              />
            </View>
          </View>
        </View>
        <MayoButton />
      </ImageBackground>
    </Theme>
  );
};
