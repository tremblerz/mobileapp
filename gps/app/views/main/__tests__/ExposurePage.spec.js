import { render } from '@testing-library/react-native';
import React from 'react';

import { ExposurePage } from '../ExposurePage';

it('may be exposed matches snapshot', () => {
  const { asJSON } = render(<ExposurePage />);

  expect(asJSON()).toMatchSnapshot();
});
