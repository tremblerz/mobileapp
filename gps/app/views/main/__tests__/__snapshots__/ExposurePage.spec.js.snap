// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`may be exposed matches snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../shared/assets/images/backgroundAtRisk.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "resizeMode": "contain",
          "right": 0,
          "top": "-13%",
        }
      }
    >
      <SvgXml
        height={1334}
        width={1334}
        xml="<svg width=\\"850\\" height=\\"850\\" viewBox=\\"0 0 850 850\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"85.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"149.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"217.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"283.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"352.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"424.5\\" stroke=\\"white\\"/>
<path d=\\"M425 465C447.091 465 465 447.091 465 425C465 402.909 447.091 385 425 385C402.909 385 385 402.909 385 425C385 447.091 402.909 465 425 465Z\\" fill=\\"white\\"/>
<path d=\\"M425 457C442.673 457 457 442.673 457 425C457 407.327 442.673 393 425 393C407.327 393 393 407.327 393 425C393 442.673 407.327 457 425 457Z\\" fill=\\"#FFC000\\"/>
<path d=\\"M424 429.375C423.5 429.5 423.25 430 423.25 430.5C423.25 431.125 423.75 431.75 424.5 431.75C425.25 431.75 425.75 431.25 425.75 430.5C425.75 429.5 424.75 429 424 429.375Z\\" fill=\\"white\\"/>
<path d=\\"M424.5 419.25C423.75 419.25 423.25 419.75 423.25 420.5V426.75C423.25 427.5 423.75 428 424.5 428C425.25 428 425.75 427.5 425.75 426.75V420.5C425.75 419.75 425.25 419.25 424.5 419.25Z\\" fill=\\"white\\"/>
<path d=\\"M424.5 413C417.625 413 412 418.625 412 425.5C412 432.375 417.625 438 424.5 438C431.375 438 437 432.375 437 425.5C437 418.625 431.375 413 424.5 413ZM424.5 435.5C419 435.5 414.5 431 414.5 425.5C414.5 420 419 415.5 424.5 415.5C430 415.5 434.5 420 434.5 425.5C434.5 431 430 435.5 424.5 435.5Z\\" fill=\\"white\\"/>
</svg>"
      />
    </View>
    <View
      style={
        Object {
          "height": "100%",
          "left": 0,
          "paddingBottom": 12,
          "paddingHorizontal": "12%",
          "position": "absolute",
          "right": 0,
          "top": "-10%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-end",
            "paddingBottom": 40,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Medium",
              "fontSize": 28,
              "fontWeight": "normal",
              "lineHeight": 34,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          You May Be Exposed
        </Text>
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Light",
              "fontSize": 16,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          This does not mean you are infected.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingTop": 80,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans",
              "fontSize": 18,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          Based on your GPS history, it is possible you were in contact with or close to someone diagnosed with COVID-19.
        </Text>
        <View
          style={
            Object {
              "height": 54,
              "marginTop": 24,
            }
          }
        >
          <View
            accessibilityLabel="See exposure history"
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#FFF",
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderColor": "transparent",
                "borderStyle": "solid",
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 54,
                "justifyContent": "center",
                "marginTop": 24,
                "opacity": 1,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3C475B",
                  "fontFamily": "IBMPlexSans-Medium",
                  "fontSize": 20,
                  "fontWeight": "normal",
                  "lineHeight": 40,
                  "writingDirection": "ltr",
                }
              }
              use="body1"
            >
              See exposure history
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "flex-end",
              "flexDirection": "column",
              "justifyContent": "space-between",
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            testID="MayoLinkHeading"
            use="body1"
          >
            More COVID-19 information
          </Text>
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            testID="MayoLinkLabel"
            use="body1"
          >
            from the Mayo Clinic
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../shared/assets/images/foreArrow.png",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
